File: agility-game-core/src/main/java/org/rnd/agility/game/domain/join/dto/GameThumbnail.java
package org.rnd.agility.game.domain.join.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Getter;
import org.rnd.agility.game.domain.game.Game;

@Getter
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class GameThumbnail {
    private String gameId;
    private String status;
    private Integer size;

    public GameThumbnail(Game game){
        this.gameId = game.getGameId();
        this.status = game.getStatus().get().getValue();
        this.size = game.getUsers().size();
    }
}

File: agility-game-view/src/components/GameThumbnail.js
import React from 'react'
const GameThumbnail = (props) => {

    const handleClick = (e) => {
        props.tryJoinGame(props.game.game_id)
    }

    return (
        <tr className="GameThumbnail" onClick={handleClick}>
            <td>{props.game.game_id}</td>
            <td>{props.game.status}</td>
            <td>{props.game.size}</td>
        </tr>
    )
}

export default GameThumbnail